SetupDataPkg/Docs/ConfigurationFiles/ConfigurationFiles.md

```diff
@@ -32,122 +31,184 @@ This document is intended to describe the Project MU version of Configuration Fi
 | ----------------------------------------- | ----------------------------------- |
-| Slim Bootloader Repo | <https://github.com/slimbootloader/slimbootloader> |
-| Configuration YAML Spec | <https://slimbootloader.github.io/specs/config.html#configuration-description-yaml-explained> |
-| Project Mu Document | <https://microsoft.github.io/mu/> |
-| Configuration Apps Repo | <https://windowspartners.visualstudio.com/MSCoreUEFI/_git/mu_config_apps> |
+| Project Mu Documentation | <https://microsoft.github.io/mu/> |
 
 ## Introduction
 
-As Project MU inherits tool set from Slim Bootloader repository to support setup variable feature, certain modifications
-has been made to improve workflow and architectural abstraction.
+mu_feature_config uses a specific XML schema to describe configuration for a platform. From this XML, change files,
+variable list binaries, and SVD files can be generated to manipulate settings. This file describes the XML format.
 
-Although the syntax of configuration YAML files mainly follow Slim Bootloader specification for design simplicity, this
-document mainly listed the differences between Project MU version and original Slim Bootloader. Additionally, this document
-describes the XML format that will be accepted.
+## XML Specification
 
-## YAML Specification Differences
+See [sampleschema.xml](../../Tools/sampleschema.xml) for an example XML schema.
 
-- All UI related form fields (i.e. `name`, `type`, `help`, etc.) should be described separately from the data definitions.
+Configuration will be organized in namespaces, each consisting of various knobs. Knobs may be built of children knobs
+or be a leaf knob.
 
-- The UI form set should be named as `*_UI` following the same directory/name of data form sets (see example below). The
-intention of this separation is to allow cleaner YAML layout for platform configuration template creation while maintaining
-the [ConfigEditor.py](../../Tools/ConfigEditor.py) capability of updating configuration offline.
+Supported data types (and their mapping in UEFI) are:
+
+- uint8_t   (UINT8)
+- int8_t    (INT8)
+- uint16_t  (UINT16)
+- int16_t   (INT16)
+- uint32_t  (UINT32)
+- int32_t   (INT32)
+- uint64_t  (UINT64)
+- int64_t   (INT64)
+- float     (float)
+- double    (double)
+- bool      (BOOLEAN)
+
+In addition, user defined enums and structs can be created. Structs will be packed and enums resolve to a series of
+INT32s.
+
+### User Defined Enums
+
+A `Enums` block can be added to define enums (which resolve to INT32s) for a platform, as follows:
+
+```xml
+  <Enums>
+    <Enum name="OPTION_MODE" help="Modes of the option">
+      <Value name="FIRST" value="0" help="First mode" />
+      <Value name="SECOND" value="1" help="Second mode" />
+      <Value name="THIRD" value="2" help="Third mode" />
+    </Enum>
+  </Enums>
+```
 
-```bash
-  CfgDataDef.yaml
-  CfgDataDef_UI.yaml
-  | Template_USB.yaml
-  | Template_USB_UI.yaml
+### User Defined Structs

```

Comment:
Not sure if this needs to be detailed here, but is there a defined packing for these structs? In case they are consumes across multiple compiler environments.

Reply:
Added pointer to packing definition.